lib/paramedic.js

```diff
@@ -175,39 +178,58 @@ ParamedicRunner.prototype.runTests = function() {
             reject(new Error('device is disconnected before passing the tests'));
         });
 
-        var command = self.getCommandForStartingTests();
-        logger.normal('cordova-paramedic: running command ' + command);
-
-        exec(command, function(code, output) {
-            if(code) {
-                // this trace is automatically available in verbose mode
-                // so we check for this flag to not trace twice
-                if (!self.config.verbose) {
-                    logger.normal(output);
+        return self.getCommandForStartingTests()
+        .then(function(command){
+            self.setPermissions();
+            logger.normal('cordova-paramedic: running command ' + command);
+            exec(command, function(code, output) {
+                if(code) {
+                    // this trace is automatically available in verbose mode
+                    // so we check for this flag to not trace twice
+                    if (!self.config.verbose) {
+                        logger.normal(output);
+                    }
+                    logger.normal('cordova-paramedic: unable to run tests; command log is available above');
+                    return reject(new Error(command + " returned error code " + code));
                 }
-                logger.normal('cordova-paramedic: unable to run tests; command log is available above');
-                return reject(new Error(command + " returned error code " + code));
-            }
 
-            // skip tests if it was just build
-            if (!self.shouldWaitForTestResult()) {
-                return resolve(true);
-            }
+                // skip tests if it was just build
+                if (!self.shouldWaitForTestResult()) {
+                 return resolve(true);
```

Comment:
@sarangan12 the `return` statement is only indented by one space. Should be indented by four spaces